File: java/search/HashTableSearch.java
package search;

import java.util.HashMap;

public class HashTableSearch {
    public static void main(String[] args) {
        // Set con datos de prueba
        HashMap<Integer, String> hashTable = new HashMap<>();
        hashTable.put(5, "cinco");
        hashTable.put(2, "dos");
        hashTable.put(9, "nueve");
        hashTable.put(1, "uno");
        hashTable.put(6, "seis");
        
        // Existe
        int keyToFind = 9;
        boolean exists = hashTableSearch(hashTable, keyToFind);
        System.out.println("Existe la clave " + keyToFind + "? " + exists);
        
        // No existe
        keyToFind = 7;
        exists = hashTableSearch(hashTable, keyToFind);
        System.out.println("Existe la clave " + keyToFind + "? " + exists);
    }
    
    public static boolean hashTableSearch(HashMap<Integer, String> hashTable, int key) {
        return hashTable.containsKey(key);
    }
}

File: python/search/hash_table_search.py
def hash_table_search(hash_table, key):
    return key in hash_table

if __name__ == "__main__":
    # Diccionario con datos de prueba
    hash_table = {
        5: "cinco",
        2: "dos",
        9: "nueve",
        1: "uno",
        6: "seis"
    }
    
    # Existe
    key_to_find = 9
    exists = hash_table_search(hash_table, key_to_find)
    print(f"¿Existe la clave {key_to_find}? {exists}")
    
    # No existe
    key_to_find = 7
    exists = hash_table_search(hash_table, key_to_find)
    print(f"¿Existe la clave {key_to_find}? {exists}")
